src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeSummaryRepositoryTest.java

```diff
@@ -0,0 +1,203 @@
+
+        // after the request date
+        envelopeRepository.insert(new NewEnvelope(CONTAINER_2, FILE_2_2, createdAt4, dispatchedAt, Status.REJECTED));
+        eventRecordRepository.insert(new NewEventRecord(CONTAINER_2, FILE_2_2, FILE_PROCESSING_STARTED));
+
+        // when
+        List<EnvelopeSummary> result = envelopeSummaryRepository.find(
+            LocalDate.parse("2019-12-20").atStartOfDay().toInstant(UTC),
+            LocalDate.parse("2019-12-21").atStartOfDay().toInstant(UTC)
+        );
+
+        // then
+        assertThat(result.stream()
+                       .collect(toList()))
+            .usingFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                new EnvelopeSummary(
+                    CONTAINER_1,
+                    FILE_1_2,
+                    createdAt2,
+                    dispatchedAt,
+                    Status.DISPATCHED,
+                    false,
+                    DISPATCHED,
+                    null,
+                    lastEvent12CreatedAt
+                ),
+                new EnvelopeSummary(
+                    CONTAINER_2,
+                    FILE_2_1,
+                    createdAt3,
+                    null,
+                    Status.REJECTED,
+                    false,
+                    DUPLICATE_REJECTED,
+                    "Duplicate",
+                    lastEvent21CreatedAt
+                )
+            );
+    }
+
+    @Test
+    @SuppressWarnings("checkstyle:variabledeclarationusagedistance")
+    void should_handle_missing_events() {
+        // given
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        Instant createdAt = LocalDateTime.parse("2019-12-20 10:31:25", formatter).toInstant(UTC);
+        Instant dispatchedAt = LocalDateTime.parse("2019-12-22 13:39:33", formatter).toInstant(UTC);
+
+        envelopeRepository.insert(new NewEnvelope(CONTAINER_1, FILE_1_1, createdAt, dispatchedAt, Status.DISPATCHED));
+
+        // when
+        List<EnvelopeSummary> result = envelopeSummaryRepository.find(
+            LocalDate.parse("2019-12-20").atStartOfDay().toInstant(UTC),
+            LocalDate.parse("2019-12-21").atStartOfDay().toInstant(UTC)
+        );
+
+        // then
+        assertThat(result.stream()
+                       .collect(toList()))
+            .usingFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                new EnvelopeSummary(
+                    CONTAINER_1,
+                    FILE_1_1,
+                    createdAt,
+                    dispatchedAt,
+                    Status.DISPATCHED,
+                    false,
+                    null,
+                    null,
+                    null
+                )
+            );
+    }
+
+    @Test
+    @SuppressWarnings("checkstyle:variabledeclarationusagedistance")
+    void should_skip_bulkscan_repository() {
```

Comment:
what's tested here?

Reply:
Envelopes from bulkscan repository should not be included in reports.

Reply:
you mean 'container'?

Reply:
@rkondratowicz Yes, renamed the method